File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0 ; r = len(numbers) - 1
        res = []
        
        while l<r:
            sum = numbers[l] + numbers[r]
            if target == sum:
                res.append(l+1)
                res.append(r+1)
                return res 
            elif target < sum:
                r-=1
            elif target > sum:
                l+=1

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0 ; int r = numbers.length-1;
        while(l<r){
            int sum =  numbers[l] + numbers[r];
            if(target==sum){
                return new int[]{l+1,r+1};
            }
            else if(target < sum){
                r-=1;
            }
            else{
                l+=1;

            }
        }
    return new int[]{-1,-1};

    }

}